Core/ControlMapper.cpp

```diff
@@ -49,24 +50,48 @@ void ControlMapper::SetRawCallback(std::function<void(int, float, float)> setRaw
 	setRawAnalog_ = setRawAnalog;
 }
 
-
-void ControlMapper::SetPSPAxis(char axis, float value, int stick) {
-	static float history[2][2] = {};
-
+void ControlMapper::SetPSPAxis(int device, char axis, float value, int stick) {
 	int axisId = axis == 'X' ? 0 : 1;
 
-	history[stick][axisId] = value;
+	float position[2];
+	position[0] = history[stick][0];
+	position[1] = history[stick][1];
 
-	float x = history[stick][0];
-	float y = history[stick][1];
+	position[axisId] = value;
+
+	float x = position[0];
+	float y = position[1];
 
 	if (setRawAnalog_) {
 		setRawAnalog_(stick, x, y);
 	}
 
-	ConvertAnalogStick(x, y);
+	// NOTE: We need to use single-axis checks, since the other axis might be from another device,
+	// so we'll add a little leeway.
+	bool inDeadZone = fabsf(value) < g_Config.fAnalogDeadzone * 0.7f;
+
+	bool ignore = false;
+
+	if (inDeadZone && lastNonDeadzoneDeviceID_[stick] != device) {
+		// Ignore this event! See issue #15465
+		NOTICE_LOG(COMMON, "Ignoring deadzone event from device %d (%d, %f)", device, axis, value);
+		ignore = true;
+	}
+
+	if (!inDeadZone) {
+		NOTICE_LOG(COMMON, "Got a non deadzone event from device %d (%d, %f > %f)", device, axis, value, g_Config.fAnalogDeadzone);
```

Comment:
This also seems kinda spammy to be NOTICE?

-[Unknown]

Reply:
Yeah, removing.